core/notification/handler.go

```diff
@@ -0,0 +1,120 @@
+package notification
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/odpf/salt/log"
+	"github.com/odpf/siren/pkg/errors"
+)
+
+const (
+	defaultPollDuration = 5 * time.Second
+	defaultBatchSize    = 1
+)
+
+// Handler is a process to handle message publishing
+type Handler struct {
```

Comment:
```suggestion
type Worker struct {
```
`Worker` seems more accurate to the usecase

Reply:
I think `Worker` is too generic and would give more confusion. `Handler` would be more suitable from the functionality point of view.